### DISCUSSION TITLE
WhenExpressions supporting OR conditions

### DISCUSSION THREAD

**Discussion by gdisanto** [tektoncd/pipeline#3927] [Ideas]
Hi,
Sometimes in my pipelines I feel the need to specify complex when conditions. For example running a task if ( the branch is develop AND a new commit has been pushed) OR (the branch is master and a new tag was created).
As far as I understood when expressions support only AND for now. My only solution at the moment is to create two tasks and apply each condition branch to one of them. It works but it is not the most elegant solution I would say.
Are there any plans to improve the when expressions in the future?.

Thanks,
Giovanni

**Comment by jerop** [comment]
👋🏾 

you can use common expression language to evaluate complex expressions through custom tasks -- the evaluation results can then be used in whenexpressions in subsequent tasks: https://github.com/tektoncd/experimental/tree/main/cel

that should work for the use case you described above, please share any feedback on cel custom tasks as we're iterating on it 

to add more context, we want to keep whenexpressions simple and delegate the complex expressions to an expression language ([don't want to implement our own expression language](https://github.com/tektoncd/community/blob/main/design-principles.md#flexibility))

**Reply by gdisanto** [reply]
Hi @jerop,
Thanks a lot for your reply. I'll try what you are suggesting.
I have also noticed that CEL expression can be used on triggers filters as well. I'll probably try this option as it means that the pipelinerun won't be instantiated at all and that would make sense, at least on my case.

Thanks again,
Giovanni

**Reply by jerop** [reply]
sounds great, please us know how it goes with using cel custom tasks 😀
